### ISSUE TITLE
Generator for StyledProperty does not work anymore on Avalonia 11.0.0 preview6

### ISSUE THREAD

**Issue by YBAZAN** [HavenDV/DependencyPropertyGenerator#11]
### Describe the bug

In the latest pre-release of Avalonia 11 (preview 6), the prototype of the Register method with the parameter `notifying` is now internal.

Here a discussion on the Avalonia repo explaining the change: https://github.com/AvaloniaUI/Avalonia/discussions/10766

So your library cannot be used with this version and the later ones.

### Steps to reproduce the bug

1. Migrate the Avalonia nuget references to `11.0.0-preview6`

The project no longer builds.

**Comment by HavenDV** [comment]
I'm not entirely sure if the other changes will apply to older versions as well. There are two ways here - to introduce the MSBuild property DependencyPropertyGenerator_UseAvaloniaPreview, which the generator will read and generate the correct code, or wait until these versions come out of preview. I will implement the first option if there are messages from you or other users that this is necessary.

**Comment by HavenDV** [comment]
Message from Intellisence or message when compiling?

**Comment by YBAZAN** [comment]
When compiling

**Comment by YBAZAN** [comment]
Is it something related to https://github.com/HavenDV/H.Generators.Extensions/blob/108ce2b09a42ddfd1573c78c82617419fa58df47/src/libs/H.Generators.Extensions/AnalyzerConfigOptionsProviderExtensions.cs#L141

Do I need to set particular settings in my csproj (RecognizeFramework_DefinedConstants...)?

**Comment by HavenDV** [comment]
Yes, try setting `<RecognizeFramework_DefineConstants>HAS_AVALONIA</RecognizeFramework_DefineConstants>` explicitly. I still don't understand what the problem is

**Comment by HavenDV** [comment]
Perhaps your build fails for another reason, but you see not the build log, but Intellisence errors that may not be correct?

**Comment by YBAZAN** [comment]
I've found the problem. I didn't add a reference to the DependencyPropertyGenerator package (+ DefineConstants) in the project. I thought it would consume the package as transient from a referenced project.

It was working until now, so I didn't what changed since. 

BTW, it works, I'm happy !

Thank you very much for the time you've spent to help me, very appreciated :)
